services/dqa-secondary-reports/main.py

```diff
@@ -0,0 +1,271 @@
+
+                    self.table_totals.setdefault(version, {})
+                    self.field_totals.setdefault(version, {})
+
+                    for file_name in file_names:
+                        table = file_name.split('.csv')[0]
+                        buff = codecs.open(
+                            os.path.join(etl_path, file_name), 'r',
+                            encoding='latin1')
+                        p = Parser(buff, hospital, table,
+                                   self.field_totals[version],
+                                   self.table_totals[version],
+                                   self.site_totals[version][hospital])
+                        p.parse()
+
+    def process_dictionary(self):
+        path = './dqa_repo/Dictionary/DCC_DQA_Dictionary.csv'
+        buff = codecs.open(path, 'r', encoding='latin1')
+        self.dictionary = DictionaryParser(buff).parse()
+
+    def update(self):
+        self.update_repo()
+        self.process_dqa()
+        self.process_dictionary()
+
+    def update_repo(self):
+        logger.debug('start updating repo')
+
+        needs_refresh = True
+
+        if os.path.isdir(self.dir_name):
+            try:
+                git.Repo(self.dir_name).remotes.origin.pull()
+                needs_refresh = False
+            except:
+                shutil.rmtree(self.dir_name)
+
+        if needs_refresh:
+            os.mkdir(self.dir_name)
+            git.Repo.clone_from(self.remote_url, self.dir_name)
+
+        logger.debug('done updating repo')
+
+    def get_field_totals(self, version):
+        "Entrypoint for Flask routing for a given version of PEDSnet data model"
+        def versioned_data():
+            try:
+                info = self.field_totals[version]
+            except HTTPError:
+                return 503, ''
+
+            return self.wrap_response(info)
+
+        return versioned_data
+
+    def get_dict(self):
+        "Entrypoint for Flask routing for dictionary of DQA status codes"
+        try:
+            info = self.dictionary
+        except HTTPError:
+            return 503, ''
+
+        return self.wrap_response(info)
+
+    def get_site_totals(self, version):
+        "Entrypoint for Flask routing for site totals for a given version of PEDSnet data model"
+        def versioned_data():
+            try:
+                info = self.site_totals[version]
+            except HTTPError:
+                return 503, ''
+
+            return self.wrap_response(info)
+
+        return versioned_data
+
+    def get_table_totals(self, version):
+        "Entrypoint for Flask routing for table totals for a given version of PEDSnet data model"
+        def versioned_data():
+            try:
+                info = self.table_totals[version]
+            except HTTPError:
+                return 503, ''
+
+            return self.wrap_response(info)
+
+        return versioned_data
+
+    def wrap_response(self, info):
```

Comment:
This is a utility function that can be moved out of the class (it does not use `self` at all).